membership-attribute-service/app/controllers/ExistingPaymentOptionsController.scala

```diff
@@ -81,7 +81,7 @@ class ExistingPaymentOptionsController(commonActions: CommonActions, override va
     val defaultMandateIdIfApplicable = "CLEARED"
 
     def paymentMethodStillValid(paymentMethodOption: Option[PaymentMethod]) = paymentMethodOption match {
-      case Some(card: PaymentCard) => card.paymentCardDetails.exists(cardThatWontBeExpiredOnFirstTransaction)
+      case Some(card: PaymentCard) => card.isReferenceTransaction && card.paymentCardDetails.exists(cardThatWontBeExpiredOnFirstTransaction)
```

Comment:
looks good re the changes.
also I know it's old code but in the original code I'd probably pass in the PaymentMethod itself rather than an Option[PaymentMethod] to the functions here, as the functions wouldn't do anything in the case of a None so it's an extra input that's not needed.  This would also make unit testing easier (if members-data-api had them)